### ISSUE TITLE
backend service needs to omit option for balancing_mode

### ISSUE THREAD

**Issue by TobiasLierzer** [hashicorp/terraform-provider-google#17257]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Terraform Version

Terraform v1.7.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v5.15.0
+ provider registry.terraform.io/hashicorp/google-beta v5.15.0

### Affected Resource(s)

google_compute_region_backend_service
google_compute_region_network_endpoint_group

### Terraform Configuration

```tf
resource "google_compute_region_network_endpoint_group" "default" {
  name    = "public-endpoint-neg"
  region  = "europe-west4"
  network = local.network

  network_endpoint_type = "INTERNET_FQDN_PORT"
}
resource "google_compute_region_backend_service" "default" {
  name     = "public-backend"
  region   = "europe-west4"
  protocol = "TCP"

  backend {
    group = google_compute_region_network_endpoint_group.default.id
    balancing_mode = null
  }
  load_balancing_scheme = "INTERNAL_MANAGED"
}
```

### Debug Output

Step #1 - "tf-apply": google_compute_region_backend_service.default: Creating...
Step #1 - "tf-apply": 
Step #1 - "tf-apply": Error: Error creating RegionBackendService: googleapi: Error 400: Invalid value for field 'resource.backends[0].balancingMode': 'CONNECTION'. Balancing mode cannot be set for an Internet Network Endpoint Group backend., invalid
Step #1 - "tf-apply": 
Step #1 - "tf-apply":   with google_compute_region_backend_service.default

### Expected Behavior

The backend_service should consider that balancing_mode is set to null and create the backend_service accordingly without balancing_mode set

### Actual Behavior

As per google documentation on loadbalancer balancing modes [1], internet NEG in conjunction with Proxy Network Loadbalancers need to omit a value for balancing mode. 
As per terraform documentation [2], "Possible values are: UTILIZATION, RATE, CONNECTION."
Setting balancing_mode to "null" as seen in the sample above does not take effect. The default CONNECTION is used in that case.

[1] https://cloud.google.com/load-balancing/docs/backend-service#balancing-mode-lb
[2] https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_backend_service#balancing_mode

### Steps to reproduce

1. `terraform apply`

**Comment by TobiasLierzer** [comment]
https://github.com/hashicorp/terraform-provider-google/issues/12299 describes a similar issue with a different backend. I'll test the given solution (setting balancing_mode to UTILIZATION) but this does not match the official documentation

**Comment by edwardmedia** [comment]
Currently [default](https://github.com/hashicorp/terraform-provider-google/blob/fee6c6085f8cf67a2b4d074dadb5e12fc9dcbe1b/google/services/compute/resource_compute_region_backend_service.go#L950) is given to balancing_mode. It looks like the api has updated it behavior. 

Based on the [doc](https://cloud.google.com/load-balancing/docs/backend-service#connection_balancing_mode), there are cases where `balancing mode must be omitted`

```
The Application Load Balancers support either RATE or UTILIZATION balancing modes for instance group backends, RATE balancing mode for zonal NEGs with GCE_VM_IP_PORT endpoints, and RATE balancing mode for hybrid NEGs (NON_GCP_PRIVATE_IP_PORT endpoints). For any other type of supported backend, balancing mode must be omitted.
```

**Comment by macwas** [comment]
@TobiasLierzer, you are right that for Internet NEGs, load-balancing-mode should be omitted and Terraform currently disallows it.

However, as a short-term solution, could you try to set it UTILIZATION? We believe that it might work for Internet NEGs. In the meantime we will be working on a long-term solution.

**Comment by TobiasLierzer** [comment]
Hi macwas,

setting the mode to UTILIZATION worked. I am able to deploy the backend_service with that.

**Comment by pawelJas** [comment]
The default value for balancing_mode in `google_compute_region_backend_service` has been changed to `UTILIZATION` (default GCP value) in version 6.0.0.

**Comment by billyfoss** [comment]
I am seeing "UTILIZATION" being set as a default in 5.43.0 and that triggers the following error. 

googleapi: Error 400: Invalid value for field 'resource.backends[0].balancingMode': 'UTILIZATION'. Balancing mode must be CONNECTION for an INTERNAL backend service.

The plan is trying to change the existing CONNECTION to UTILIZATION
````

   # google_compute_region_backend_service.default will be updated in-place
   ~ resource "google_compute_region_backend_service" "default" {
         id                              = "projects/REDACTED/regions/us-east1/backendServices/lb-REDACTED"
         name                            = "lb-REDACTED"
         # (14 unchanged attributes hidden)
 
       - backend {
           - balancing_mode               = "CONNECTION" -> null
           - capacity_scaler              = 0 -> null
           - failover                     = false -> null
           - group                        = "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-east1-b/instanceGroups/lb-REDACTED-us-east1-b-lb-instance-group" -> null
           - max_connections              = 0 -> null
           - max_connections_per_endpoint = 0 -> null
           - max_connections_per_instance = 0 -> null
           - max_rate                     = 0 -> null
           - max_rate_per_endpoint        = 0 -> null
           - max_rate_per_instance        = 0 -> null
           - max_utilization              = 0 -> null
         }
       - backend {
           - balancing_mode               = "CONNECTION" -> null
           - capacity_scaler              = 0 -> null
           - failover                     = false -> null
           - group                        = "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-east1-c/instanceGroups/lb-REDACTED-us-east1-c-lb-instance-group" -> null
           - max_connections              = 0 -> null
           - max_connections_per_endpoint = 0 -> null
           - max_connections_per_instance = 0 -> null
           - max_rate                     = 0 -> null
           - max_rate_per_endpoint        = 0 -> null
           - max_rate_per_instance        = 0 -> null
           - max_utilization              = 0 -> null
         }
       + backend {
           + balancing_mode = "UTILIZATION"
           + failover       = (known after apply)
           + group          = "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-east1-b/instanceGroups/lb-REDACTED-us-east1-b-lb-instance-group"
         }
       + backend {
           + balancing_mode = "UTILIZATION"
           + failover       = (known after apply)
           + group          = "https://www.googleapis.com/compute/v1/projects/REDACTED/zones/us-east1-c/instanceGroups/lb-REDACTED-us-east1-c-lb-instance-group"
         }
 
       - iap {
           - enabled                     = false -> null
           - oauth2_client_secret_sha256 = (sensitive value) -> null
         }
     }
 ````

**Comment by billyfoss** [comment]
@pawelJas  Yes, I believe my config had updated to the 6.x provider. 

I have updated a test environment to 6.7.0 and I am seeing the errors again.  I am not sure if this should be a separate ticket.  The root cause is common and possibly opposite of the original ticket.  

In the 5.x provider, the default is CONNECTION
In the 6.x provider, the default is UTILIZATION

However, according to https://cloud.google.com/load-balancing/docs/backend-service#balancing-mode-lb
UTILIZATION is not valid for several LB types.  So the new default is triggering existing resources to see the following error. 

Error 400: Invalid value for field 'resource.backends[0].balancingMode': 'UTILIZATION'. Balancing mode must be CONNECTION for an INTERNAL backend service.

**Comment by pawelJas** [comment]
Hey @billyfoss,

GCP default value is UTILIZATION. Not providing any BalancingMode in Terraform is the same as providing UTILIZATION so it makes more sense to assume default UTILIZATION in Terraform. That was the reason for a my change.

Any default enum value will cause a problem in some configuration. Major version upgrades are not backward compatible and require manual intervention. In the context of BalancingMode setting CONNECTION everywhere where it was omitted to avoid errors as above.

**Comment by danhiris** [comment]
when I specify CONNECTION with UNSPECIFIED it no longer picks up the balancing_mode. It still sets it to the default of UTILIZATION
